WordPress/Classes/ViewRelated/Me/All Domains/Views/AllDomainsListViewController.swift

```diff
@@ -0,0 +1,149 @@
+        // Setup navigation item
+        self.navigationItem.rightBarButtonItem = .init(systemItem: .add)
+
+        // Setup search bar
+        let searchController = UISearchController(searchResultsController: nil)
+        searchController.delegate = self
+        searchController.searchBar.delegate = self
+        searchController.searchBar.placeholder = Strings.searchBar
+        self.navigationItem.searchController = searchController
+        self.navigationItem.hidesSearchBarWhenScrolling = false
+        self.extendedLayoutIncludesOpaqueBars = true
+        self.edgesForExtendedLayout = .top
+
+        // Setup tableView
+        self.tableView.translatesAutoresizingMaskIntoConstraints = false
+        self.tableView.dataSource = self
+        self.tableView.delegate = self
+        self.tableView.sectionHeaderHeight = .leastNormalMagnitude
+        self.tableView.sectionFooterHeight = Layout.interRowSpacing
+        self.tableView.contentInset.top = Layout.interRowSpacing
+        self.tableView.register(AllDomainsListTableViewCell.self, forCellReuseIdentifier: CellIdentifiers.myDomain)
+        self.tableView.register(AllDomainsListActivityIndicatorTableViewCell.self, forCellReuseIdentifier: CellIdentifiers.activityIndicator)
+        self.view.addSubview(tableView)
+        self.view.pinSubviewToAllEdges(tableView)
+    }
+
+    private func observeState() {
+        self.viewModel.$state.sink { [weak self] state in
+            guard let self else {
+                return
+            }
+            self.state = state
+            switch state {
+            case .normal, .loading:
+                self.tableView.reloadData()
+            case .error:
+                break
+            case .empty:
+                break
+            }
+        }.store(in: &cancellable)
+    }
+
+    // MARK: - Types
+
+    private enum Layout {
+        static let interRowSpacing = Length.Padding.double
+    }
+
+    private enum CellIdentifiers {
+        static let myDomain = String(describing: AllDomainsListTableViewCell.self)
+        static let activityIndicator = String(describing: AllDomainsListActivityIndicatorTableViewCell.self)
+    }
+
+    typealias ViewModel = AllDomainsListViewModel
+}
+
+// MARK: - UITableViewDataSource
+
+extension AllDomainsListViewController: UITableViewDataSource, UITableViewDelegate {
+
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        // Workaround to accurately control section height using `tableView.sectionHeaderHeight`.
+        return UIView()
+    }
+
+    func tableView(_ tableView: UITableView, viewForFooterInSection section: Int) -> UIView? {
+        // Workaround to accurately control footer height using `tableView.sectionFooterHeight`.
+        return UIView()
+    }
+
+    func numberOfSections(in tableView: UITableView) -> Int {
+        switch state {
+        case .loading: return 1
+        default: return viewModel.numberOfDomains
```

Comment:
I understand we're doing this so we can add spacing between each card. But I believe this approach is very restrictive moving forward. Say we decide to tweak the design in the future to add actual sections, it won't be a simple task then.

I think using a collection view with a compositional layout is more fitting to our use case. Adding padding to the cell itself is also an option. Let me know what you think!

Reply:
> What do you say I reiterate on this after we wrap up the project?

Works for me 👍